controller_interface/src/controller_interface.cpp

```diff
@@ -28,8 +28,7 @@ ControllerInterface::init(const std::string & controller_name)
 {
   node_ = std::make_shared<rclcpp::Node>(
     controller_name,
-    rclcpp::NodeOptions().allow_undeclared_parameters(true).
-    automatically_declare_parameters_from_overrides(true));
+    rclcpp::NodeOptions().allow_undeclared_parameters(true));
```

Comment:
If you remove the second line there were issues with not auto-declaring / auto-setting of parameters. Please check against demo repository before merging.

Reply:
That's how I actually found this issue. Anything that uses a more complex parameter setup can't work with the original set of flags.

I think this used to be here as a lucky charm when Dashing had issues with param handling. It is supposed to allow auto-declaring parameters for a node when they are set from outside but the `automatically_declare_parameters_from_overrides` flag forces those to be declared, when we re-declare it inside `init()`, there's a clash. If we want to pre-load yaml configurations for controllers that also have defaults if unspecified, we need to remove that line.

Reply:
https://github.com/ros-controls/ros2_controllers/pull/131